src/main.c

```diff
@@ -0,0 +1,49 @@
+/******************************************************
+* Author : Amuleen Gulati 
+* Function : main
+* Application : This function takes sensor values as input and 
+* calls different functions to produce an output
+* Input : a .csv file containing temperature values of sensors
+* Output : a single temperature value
+*/
+
+#include<stdio.h>
+
+#include "..\include\sdm.h"
+#include "..\include\eigen.h"
+#include "..\include\principle_comp.h"
+#include "..\include\contribution.h"
+
+int main()
+{
+int n = 4;
+double temperature[] = {2.75,3.55,3.70,4.0};
+
+double** sd_matrix;
+sd_matrix = (double**) malloc(n*sizeof(double));
+for(int r = 0; r < n; r++){
+    sd_matrix[r] = (double*) malloc(n*sizeof(double));
+}
+sd_matrix = support_degree_matrix(n,temperature);
+
+double** T;
```

Comment:
Non-descriptive variable name...

Reply:
I have created these variable names as they are in the given algorithm. So its easier to identify without any loss of generality.